src/api/event/controllers/event.js

```diff
@@ -1,9 +1,38 @@
-'use strict';
+const { createCoreController } = require('@strapi/strapi').factories;
 
-/**
- * event controller
- */
+module.exports = createCoreController('api::event.event', ({ strapi }) => ({
+  async find(ctx) {
+    // filter with the query
+    const { entityId, entityType } = ctx.query;
 
-const { createCoreController } = require('@strapi/strapi').factories;
+    // Fetch only the events that match the filters
+    const filters = {};
+    if (entityId) filters.entityId = entityId;
+    if (entityType) filters.entityType = entityType;
+
+    const events = await strapi.entityService.findMany('api::event.event', {
+      filters,  // Apply filters to fetch specific events
+      populate: ['eventUser'],  // Populate relations (e.g., eventUser)
+    });
+
+    // Retrieve latest entityName for each event Dynamically
+    const latestEvents = await Promise.all(events.map(async (event) => {
+      let entityName = '';
+
+      if (event.entityType === 'IdeaCard') {
+        const ideaCard = await strapi.entityService.findOne('api::idea-card.idea-card', event.entityId);
+        entityName = ideaCard ? ideaCard.ideaName : 'Unknown Idea';
```

Comment:
Should we return an error if we can't find the idea/comment? Alternatively, we can drop the event and log an error on the server side so we know something unexpected happened (the idea/comment is missing).